### ISSUE TITLE
Fuzzing `minimumNumberFields` generates invalid tests

### ISSUE THREAD

**Issue by tillig** [apideck-libraries/portman#624]
In v1.28.0 there were some updates to fuzzing with respect to minimum number fields. We've found that this fuzzing now generates invalid tests because fields with minimum numbers get moved from nested structures to the top object level.

For example, let's say you have an object like:

```json
{
  "container": {
    "value": 5
  }
}
```

Further, let's say that `.container.value` has a `minimum` constraint in the schema so it must be >= 0.

If you enable fuzzing on minimum number fields, the request that gets generated will look like:

```json
{
  "container": {
    "value": 5
  },
  "value": -1
}
```

The value being fuzzed gets added to the _top-level object_ rather than replacing the actual property at `.container.value`. This causes the tests to fail if you have your service validating requests because there shouldn't be a top-level `value` property.

I've put together a repro to illustrate the issue.

Here's the schema - it has a single operation `POST /repro` that takes in the object like I illustrated above (`.container.value`).

<details>
<summary>Click to expand the schema JSON</summary>

```json
{
  "components": {
    "schemas": {
      "reproObject": {
        "properties": {
          "container": {
            "$ref": "#/components/schemas/valueContainer"
          }
        },
        "required": [
          "container"
        ],
        "type": "object"
      },
      "valueContainer": {
        "properties": {
          "value": {
            "minimum": 0,
            "type": "number"
          }
        },
        "required": [
          "value"
        ],
        "type": "object"
      }
    }
  },
  "info": {
    "contact": {
      "email": "support@repro.com"
    },
    "description": "Repro for Portman.",
    "title": "Repro",
    "version": "0.0.1"
  },
  "openapi": "3.0.3",
  "paths": {
    "/repro": {
      "post": {
        "description": "Repro operation that shows an incorrect value fuzzing test.",
        "operationId": "repro",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/reproObject"
              }
            }
          }
        },
        "responses": {
          "204": {
            "description": "Success."
          },
          "400": {
            "description": "Validation error."
          }
        },
        "tags": [
          "repro"
        ]
      }
    }
  },
  "servers": [
    {
      "url": "https://api.repro.com"
    }
  ],
  "tags": [
    {
      "description": "Repro.",
      "name": "repro"
    }
  ]
}
```

</details>

Here's the `portman-config.json`. It only has fuzzing enabled, and only for minimum number fields.

<details>
<summary>Click to expand the Portman configuration JSON</summary>

```json
{
  "tests": {
    "contractTests": [
    ],
    "variationTests": [
      {
        "openApiOperation": "*::/*",
        "openApiResponse": "400",
        "variations": [
          {
            "fuzzing": [
              {
                "requestBody": [
                  {
                    "minimumNumberFields": {
                      "enabled": true
                    }
                  }
                ]
              }
            ],
            "name": "400 - Validation",
            "tests": {
            }
          }
        ]
      }
    ]
  },
  "version": 1
}
```

</details>

Running `portman -l schema.json -b http://localhost:8080 -o output.json -c portman-config.json` is what I'm doing to generate the tests (with my `package.json` referencing v1.28.x).

<details>
<summary>Click to see the package.json</summary>

```json
{
  "devDependencies": {
    "@apideck/portman": "1.28.x"
  },
  "license": "ISC",
  "name": "specification",
  "scripts": {
    "build": "portman -l schema.json -b http://localhost:8080 -o output.json -c portman-config.json"
  },
  "type": "commonjs",
  "version": "1.0.0"
}
```

</details>

Here's a snippet showing the test that gets generated - note I trimmed out irrelevant bits for illustration purposes, but the raw body is the important part here:

```json
{
  "name": "repro[400 - Validation][minimum number value value]",
  "request": {
    "body": {
      "raw": "{\n    \"container\": {\n        \"value\": 55576435.36133539\n    },\n    \"value\": -1\n}"
    },
    "method": "POST"
  }
}
```

Note the object has both `.container.value` _and_ `.value`. That's the problem.

I haven't had a chance to dig into this to figure out what the issue is yet, but I thought I'd document it and get a repro posted to ensure it didn't get lost.

**Comment by thim81** [comment]
Hi @tillig 

As always, we appreciate the effort and clear description with repro.

We have indeed improved fuzzing, but it was only to[ support "0" (zero) values](https://github.com/apideck-libraries/portman/commit/4a7673c3a50f2c301d6c482d477ccf2c72d717ec#diff-66850787e859f7816eadd7212b12b36440411f5fbc8a1f38a8977782702835ee). 

I'll try to make some time to debug your repro case, and figure out what it causing this.

**Comment by thim81** [comment]
@tillig Found the cause, which was linked to 0 (zero) value not triggering the nested path generation.
This [PR](https://github.com/apideck-libraries/portman/pull/625) will restore the fuzzing.

**Comment by tillig** [comment]
Oh, nice! Glad you were able to figure it out. I hadn't gotten time to jump in myself or I'd have tried to get a PR in. I'll look forward to the next release and I'll test it on my internal solutions. Thanks!

**Comment by tillig** [comment]
On my "real project" at work I now get:

```text
/Users/travis.illig/dev/spec-folder/node_modules/dot-object/index.js:105
          throw new Error(
                ^

Error: Trying to redefine `0` which is a number
    at DotObject._fill (/Users/travis.illig/dev/spec-folder/node_modules/dot-object/index.js:105:17)
    at DotObject._fill (/Users/travis.illig/dev/spec-folder/node_modules/dot-object/index.js:114:10)
    at DotObject._fill (/Users/travis.illig/dev/spec-folder/node_modules/dot-object/index.js:114:10)
    at DotObject.str (/Users/travis.illig/dev/spec-folder/node_modules/dot-object/index.js:179:10)
    at setByPath (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/utils/setByPath.js:37:16)
    at /Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/overwrites/overwriteRequestBody.js:63:50
    at Array.map (<anonymous>)
    at overwriteRequestBodyJson (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/overwrites/overwriteRequestBody.js:36:21)
    at overwriteRequestBody (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/overwrites/overwriteRequestBody.js:12:46)
    at /Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/overwrites/applyOverwrites.js:14:148
    at Array.map (<anonymous>)
    at applyOverwrites (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/overwrites/applyOverwrites.js:6:25)
    at /Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/TestSuite.js:228:40
    at Array.map (<anonymous>)
    at TestSuite.injectOverwrites (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/TestSuite.js:225:22)
    at VariationWriter.injectVariations (/Users/travis.illig/dev/spec-folder/node_modules/@apideck/portman/dist/application/VariationWriter.js:84:28)
```

So... something is still amiss. I'll see if I can work on a repro for that.

**Comment by tillig** [comment]
Here's a repro that yields the error about trying to redefine `0`. It appears it's numeric minimums in conjunction with array values.

```json
{
  "components": {
    "schemas": {
      "requestObject": {
        "description": "Holds an array of numbers.",
        "properties": {
          "numbers": {
            "description": "An array of numbers.",
            "items": {
              "minimum": -366,
              "type": "integer"
            },
            "type": "array"
          }
        },
        "type": "object"
      }
    }
  },
  "info": {
    "contact": {
      "email": "support@repro.com"
    },
    "description": "Repro.",
    "title": "Repro",
    "version": "0.0.1"
  },
  "openapi": "3.0.3",
  "paths": {
    "/repro": {
      "post": {
        "description": "Repro for error.",
        "operationId": "repro",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/requestObject"
              }
            }
          }
        },
        "responses": {
          "204": {
            "description": "Success."
          },
          "400": {
            "description": "Bad request."
          }
        },
        "tags": [
          "repro"
        ]
      }
    }
  },
  "servers": [
    {
      "description": "Repro API.",
      "url": "https://api.repro.com"
    }
  ],
  "tags": [
    {
      "description": "Reproduction test.",
      "name": "repro"
    }
  ]
}
```

**Comment by thim81** [comment]
Hi @tillig 

I m on vacation till mid August without any laptop, so it will have to wait till I m back. Hope that this is OK.

**Comment by tillig** [comment]
No problem at all. If I get a chance to get a PR in for it, I will.

**Comment by thim81** [comment]
@tillig I think you have discovered a case that was not supported before. I created a [PR](https://github.com/apideck-libraries/portman/pull/629) using your repro case.

I'll let @nicklloyd review it and potentially release it, or it will happen mid August.

**Comment by thim81** [comment]
FYI: We just released 1.291 of Portman, with the fix for handling fuzzing of array values.

**Comment by tillig** [comment]
Verified - things are looking good as of 1.29.1. Thanks!
